egs/librispeech/ASR/conformer_ctc_madam_no_warmup/conformer.py

```diff
@@ -869,7 +869,10 @@ def __init__(
             groups=channels,
             bias=bias,
         )
-        self.norm = nn.BatchNorm1d(channels)
+        # NOTE(fangjun): The process hangs when using DDP
+        # if we try to recover from CUDA OOM, so we disable
+        # batchnorm layer here.
+        #  self.norm = nn.BatchNorm1d(channels)
```

Comment:
After disabling batch norm, training with DDP can now recover from OOM in `model.forward()` as expected.

Reply:
Mm.  Hopefully with the madam optimizer, the training will still be stable without the batchnorm.  We'll have to see.  Obviously would have to compare the performance after this change.